backend/tests/v5/e2e/routes/teamspaces/teamspaces.test.js

```diff
@@ -175,6 +264,106 @@ const testGetAvatar = () => {
+			const res = await agent.get(`${route()}/?key=${testUser2.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail if the teamspace does not exist', async () => {
+			const res = await agent.get(`${route('sldkfjdl')}/?key=${userWithLicense.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test(`should return ${templates.licenceExpired.code} if the user has an expired license`, async () => {
+			const res = await agent.get(`${route(tsWithExpiredLicense.name)}/?key=${userWithExpiredLicense.apiKey}`);
+			expect(res.body).toEqual({
+				data: { used: 0, available: 0 },
+				seats: { used: 0, available: 0 },
+			});
+		});
+
+		test('should return quota if the user has a valid license', async () => {
+			const res = await agent.get(`${route()}/?key=${userWithLicense.apiKey}`)
+				.expect(templates.ok.status);
+			const collaboratorLimit = config.subscriptions?.basic?.collaborators === 'unlimited'
+				? 'unlimited' : config.subscriptions?.basic?.collaborators + userCollabs;
+			const spaceLimit = config.subscriptions?.basic?.data + licenseData;
+			expect(res.body).toEqual(
+				{
+					data: { used: 0, available: spaceLimit },
+					seats: { used: 1, available: collaboratorLimit },
+				},
+			);
+		});
+
+		test('should return quota if the user has a valid license and unlimited collaborators', async () => {
+			const res = await agent.get(`${route(tsWithLicenseUnlimitedCollabs.name)}/?key=${userWithLicenseUnlimitedCollabs.apiKey}`)
+				.expect(templates.ok.status);
+			const spaceLimit = config.subscriptions?.basic?.data + licenseData;
+			expect(res.body).toEqual(
+				{
+					data: { used: 0, available: spaceLimit },
+					seats: { used: 1, available: 'unlimited' },
+				},
+			);
+		});
+	});
+};
+
+const testRemoveTeamspaceMember = () => {
+	describe('Remove teamspace member', () => {
+		const route = (ts = tsWithUserToRemove.name,
+			username = userToRemoveFromTs.user) => `/v5/teamspaces/${ts}/members/${username}`;
+
+		test('should fail without a valid session', async () => {
+			const res = await agent.delete(route()).expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user does not have access to the teamspace', async () => {
+			const res = await agent.delete(`${route()}/?key=${userWithLicense.apiKey}`).expect(templates.teamspaceNotFound.status);
+			expect(res.body.code).toEqual(templates.teamspaceNotFound.code);
+		});
+
+		test('should fail if the user does not have admin access to the teamspace', async () => {
+			const res = await agent.delete(`${route()}/?key=${testUser2.apiKey}`).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should fail if the teamspace does not exist', async () => {
```

Comment:
Can we make this fail the same way as not authorised to do the action, so they can't just use this endpoint to find out the list of valid teamspace names

Reply:
This is coming from TeamspacePerms.isTeamspaceMember  middleware.. should I change that method to return notAuthorized ?